builtin/process_osh.py

```diff
@@ -682,4 +694,109 @@ def Run(self, cmd_val):
         return 0
 
 
+class Kill(vm._Builtin):
+    """Send a signal to a process"""
+
+    def __init__(self, job_list):
+        # type: (process.JobList) -> None
+        self.job_list = job_list
+
+    def _SignameToSignum(self, name):
+        # type: (str) -> int
+        signal_name = name.upper()
+        if signal_name.startswith("SIG"):
+            signal_name = signal_name[3:]
+        return signal_def.GetNumber(signal_name)
+
+    def _ParsePid(self, pid_arg, pid_arg_loc):
+        # type: (str, loc_t) -> int
+        if pid_arg.startswith("%"):
+            job = self.job_list.JobFromSpec(pid_arg)
+            if job is None:
+                e_usage("invalid signal specification %r" % pid_arg,
+                        pid_arg_loc)
+            else:
```

Comment:
I would remove the `else` here, since `e_usage()` is the error path, and the normal path is `job.ProcessGroupId()`